barretenberg/sol/src/honk/PoseidonParams.sol

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs
+pragma solidity >=0.8.21;
+// Field arithmetic libraries - prevent littering the code with modmul / addmul
+
+import {Fr, FrLib} from "./Fr.sol";
+
+uint256 constant T = 4;
+uint256 constant D = 5;
+uint256 constant ROUNDS_F = 8;
+uint256 constant ROUNDS_P = 56;
+uint256 constant SBOX_SIZE = 254;
+
+struct PoseidonParams {
+    Fr[T] internal_matrix_diagonal;
+    Fr[T][T] internal_matrix;
```

Comment:
In the poseidon relation, only the diagonal matrix is used, 

we probably can delete everything + the round constants i beleive they get encoded within the selectors and will not be needed as part of the verifier

Reply:
indeed, seems like I havent used my brain lol, thx for spotting